core/validator.go

```diff
@@ -0,0 +1,37 @@
+package core
+
+import (
+	"github.com/go-playground/validator/v10"
+)
+
+func register(validator *validator.Validate) {
+	validator.RegisterValidation("source", validateSourceOrDestination)
+	validator.RegisterValidation("destination", validateSourceOrDestination)
+	validator.RegisterValidation("asset", validateAsset)
+}
+
+// Validate
+func Validate(value interface{}) error {
+	validator := validator.New()
+	register(validator)
+	if err := validator.Var(value, "required,dive"); err != nil {
```

Comment:
You could register the validator at package level, or pass through the ledger though the resolver to not recreate it each time.

Reply:
The pull-request got updated following your recommendation. The validator is now injected from the ledger resolver.